go.mod

```diff
@@ -45,7 +45,7 @@ require (
 	github.com/evanphx/json-patch v5.6.0+incompatible // indirect
 	github.com/fsnotify/fsnotify v1.5.4 // indirect
 	github.com/getsentry/sentry-go v0.12.0 // indirect
-	github.com/giantswarm/app/v6 v6.6.1 // indirect
+	github.com/giantswarm/app/v6 v6.13.0 // indirect
```

Comment:
This bump in itself would close the gap used currently as a flaky feature reported here: https://github.com/giantswarm/giantswarm/issues/23453

Also this PR aims to add support for using the `extraConfigs`, however we should also rethink support for this feature via our App Platform review ticket: https://github.com/giantswarm/giantswarm/issues/20611

Reply:
To clarify, are you suggesting NOT to move on with this feature as it might be reworked/reviewed by the managed apps team?

Reply:
my 2 cents, I believe we should move on with this feature no matter the App Platform architecture changes, it's because the default apps are created by `cluster-operator` and it knows their desired state, so it should be the component that changes them. Otherwise, if changes to these App CRs (the `.spec.extraConfigs` in this case) come from a different direction, then the desired state known by the operator won't match and it will take us back to the square one. Whatever is planned for the App Platform, in this particular case it's rather a consumer of App CRs created by other platform, hence it should be this other platform who changes and supports extending App CRs it creates.

Reply:
For CAPI, the `cluster-apps-operator` creates `app-operator` and `chart-operator` only AFAIK, default apps instead are created by `default-apps-PROVIDER` Helm Chart, which is easier to extend. Hence, there is no need to extend `cluster-apps-operator`, also it may be possible for the user to edit the App CRs created by the `default-apps-PROVIDER` Helm Chart without us extending the Chart with `.spec.extraConfigs`, because Helm does server-side apply, so it shouldn't touch the fields it doesn't own, but of course this need to be verified.